src/main/java/de/unistuttgart/towercrushbackend/controller/components/WebsocketListener.java

```diff
@@ -1,25 +1,179 @@
+            }
+        });
+    }
+
+    /**
+     * This method sends all players in the lobby the new lobby list infos
+     *
+     * @param lobby lobby that gets informed
+     * @throws JsonProcessingException
+     */
+    private void sendLobbyBroadcastPlayerJoined(final String lobby) throws JsonProcessingException {
+        final Message joinLobbyMessage = new JoinLeaveLobbyMessage(lobbyManagerService.getLobby(lobby).getPlayerNames());
+        final MessageWrapper joinLobbyMessageWrapped = websocketService.wrapMessage(joinLobbyMessage, Purpose.JOIN_LOBBY_MESSAGE);
+        simpMessagingTemplate.convertAndSend(lobbyTopic + lobby, joinLobbyMessageWrapped);
+    }
+
+    /**
+     * This method adds a player to the lobby
+     *
+     * @param lobby     lobby that the player joined
+     * @param newPlayer the player that joined
+     */
+    private void addPlayerToList(final String lobby, final Player newPlayer) {
+        lobbyManagerService.createLobbyIfNotExist(lobby);
+        lobbyManagerService.addPlayer(lobby, newPlayer);
+    }
+
+    /**
+     * This method just logs if someone subscribed to a topic/queue
+     *
+     * @param event
+     */
+    @EventListener
+    private void handleSessionSubscription(final SessionSubscribeEvent event) {
+        log.info("new subscription: " + event.toString());
+    }
+
+    /**
+     * This method decides what happens if a connection is closed
+     *
+     * @param event
+     * @throws JsonProcessingException
+     */
+    @EventListener
+    private void handleSessionDisconnection(final SessionDisconnectEvent event) throws JsonProcessingException {
+        log.info("Disconnected: " + event.toString());
+        final StompHeaderAccessor sha = StompHeaderAccessor.wrap(event.getMessage());
+        if (sha.getUser().getName().equals("developer")) {
+            handleDeveloperDisconnected();
+        } else {
+            handlePlayerDisconnected(sha);
+        }
+    }
+
+    /**
+     * This method removes all infos that are no longer needed and informs the rest of the lobby
+     *
+     * @param sha
+     * @throws JsonProcessingException
+     */
+    private void handlePlayerDisconnected(final StompHeaderAccessor sha) throws JsonProcessingException {
+        final UUID playerUUID = UUID.fromString(sha.getUser().getName());
+        final String lobby = lobbyManagerService.getLobbyFromPlayer(playerUUID);
+        lobbyManagerService.removePlayerFromList(lobby, playerUUID);
+        if (lobbyManagerService.lobbyExists(lobby)) {
+            sendLobbyBroadcastPlayerJoined(lobby);
+        }
+    }
+
+    /**
+     * This method removes a developer
+     */
+    private void handleDeveloperDisconnected() {
+        developerCount--;
+        if (developerCount <= 0) {
+            log.info("no more devs online, stop infos");
+            developerCount = 0;
+            developerTask.cancel(true);
+            developerTask = null;
+        }
     }
 }
```

Comment:
```suggestion
}

```
add newline at end of file

Reply:
yes